airflow/kubernetes/pod_generator.py

```diff
@@ -298,51 +361,171 @@ def reconcile_pods(base_pod: k8s.V1Pod, client_pod: k8s.V1Pod) -> k8s.V1Pod:
+            client_pod_cp.spec.containers.extend(base_pod.spec.containers[1:])
+            merged_spec = extend_object_field(base_pod.spec, client_pod_cp.spec, 'volumes')
+            client_pod_cp.spec = merge_objects(base_pod.spec, merged_spec)
+
+        client_pod_cp.metadata = merge_objects(base_pod.metadata, client_pod_cp.metadata)
+        client_pod_cp = merge_objects(base_pod, client_pod_cp)
 
         return client_pod_cp
+
+    @staticmethod
+    def construct_pod(
+        dag_id: str,
+        task_id: str,
+        pod_id: str,
+        try_number: int,
+        date: str,
+        command: List[str],
+        kube_executor_config: Optional[k8s.V1Pod],
+        worker_config: k8s.V1Pod,
+        namespace: str,
+        worker_uuid: str
+    ) -> k8s.V1Pod:
+        """
+        Construct a pod by gathering and consolidating the configuration from 3 places:
+            - airflow.cfg
+            - executor_config
+            - dynamic arguments
+        """
+
+        dynamic_pod = PodGenerator(
+            namespace=namespace,
+            labels={
+                'airflow-worker': worker_uuid,
+                'dag_id': dag_id,
+                'task_id': task_id,
+                'execution_date': date,
+                'try_number': str(try_number),
+                'airflow_version': airflow_version.replace('+', '-'),
+                'kubernetes_executor': 'True',
+            },
+            cmds=command,
+            name=pod_id
+        ).gen_pod()
+
+        # Reconcile the pod generated by the Operator and the Pod
+        # generated by the .cfg file
+        pod_with_executor_config = PodGenerator.reconcile_pods(worker_config,
+                                                               kube_executor_config)
+        # Reconcile that pod with the dynamic fields.
+        return PodGenerator.reconcile_pods(pod_with_executor_config, dynamic_pod)
+
+    @staticmethod
+    def deserialize_model_file(api_client: ApiClient, path: str) -> k8s.V1Pod:
+        """
+        :param api_client: K8S client object
+        :param path: Path to the file
+        :return: a kubernetes.client.models.V1Pod
+        """
+        pod = None
+        with open(path) as stream:
+            if '.json' in path:
+                pod = json.load(stream)
+            elif '.yaml' in path:
+                pod = yaml.safe_load(stream)
+            elif not pod:
+                raise AirflowConfigException("Path was neither .json nor .yaml")
+
+            # pylint: disable=protected-access
+            return api_client._ApiClient__deserialize_model(pod, k8s.V1Pod)
+
+    @staticmethod
+    def deserialize_model_string(api_client: ApiClient, string: str) -> k8s.V1Pod:
+        """
+        :param api_client: K8S client object
+        :param string: a string of the deployment
+        :return: a kubernetes.client.models.V1Pod
+        """
+        try:
+            pod = json.loads(string)
```

Comment:
We don't need to do this -- JSON is to all purposes we care about a sub-set of YAML.

Reply:
So only support YAML?

Reply:
YAML has a "compact" representation that looks identical to JSON (`{}` and `[]` etc):

```
In [2]: yaml.safe_load('{"this": "is json"}')
Out[2]: {'this': 'is json'}
```

JSON is a subset of YAML, so we can pass "json" to `yaml.safe_load` and it will work. https://yaml.org/spec/1.2/spec.html?source=post_page-----1d758a069508----------------------#id2759572. About the only thing that is "valid" json but an error in YAML is this case, except that it turns out the python YAML library doesn't enforce that.